### PR TITLE
prohibit share of an encrypted folder

### PR DESCRIPTION
**Pull request by tobiasKaminsky**
- create a folder
- share it
- try to encrypt it
---> should not be allowed

Signed-off-by: tobiasKaminsky <tobias@kaminsky.me>

- [x] Tests written, or not not needed

### CODE REVIEW

**Comment by AlvaroBrey** [comment]
Fixes https://github.com/nextcloud/android/issues/11045

File: app/src/main/java/com/owncloud/android/files/FileMenuFilter.java [hunk 1]
```diff
@@ -547,4 +547,13 @@ private boolean allNotFavorites() {
         }
         return true;
     }
+
+    private boolean isShared() {
+        for (OCFile file : files) {
+            if (file.isSharedViaLink() || file.isSharedWithSharee()) {
```
Comment (AlvaroBrey) [inline app/src/main/java/com/owncloud/android/files/FileMenuFilter.java:553]: Shouldn't this also check `file.isSharedWithMe()`?

```suggestion
            if (file.isSharedWithMe() || file.isSharedViaLink() || file.isSharedWithSharee()) {
```

Comment (tobiasKaminsky) [reply app/src/main/java/com/owncloud/android/files/FileMenuFilter.java:553]: Of course!

**Review by AlvaroBrey — APPROVED** [review]

**Pull request merged**